RubberduckTests/CodeExplorer/CodeExplorerTests.cs

```diff
@@ -327,11 +328,199 @@ public void ExportModule_Cancel()
             if (parser.State.Status >= ParserState.Error) { Assert.Inconclusive("Parser Error"); }
 
             vm.SelectedItem = vm.Projects.First().Items.First().Items.First();
-            vm.ExportCommand.Execute(vm.SelectedItem);
+            vm._exportCommand.Execute(vm.SelectedItem);
 
             component.Verify(c => c.Export("C:\\Users\\Rubberduck\\Desktop\\StdModule1.bas"), Times.Never);
         }
 
+        [TestCategory("Commands")]
+        [TestMethod]
+        public void ExportProject_TestCanExecute_ExpectTrue()
+        {
+            var builder = new MockVbeBuilder();
+
+            var projectMock = builder.ProjectBuilder("TestProject1", ProjectProtection.Unprotected)
+                .AddComponent("Module1", ComponentType.StandardModule, "");
+
+            var project = projectMock.Build();
+
+            var vbe = builder.AddProject(project).Build();
+
+
+            var component1 = project.Object.VBComponents[0];
+            var module1 = component1.CodeModule;
+
+            var mockFolderBrowser = new Mock<IFolderBrowser>();
+            var mockFolderBrowserFactory = new Mock<IFolderBrowserFactory>();
+
+            var state = new RubberduckParserState(vbe.Object, new DeclarationFinderFactory());
+            var commands = new List<CommandBase>
+            {
+                new ExportAllCommand(vbe.Object, mockFolderBrowserFactory.Object)
+            };
+
+            var vm = new CodeExplorerViewModel(new FolderHelper(state), state, commands);
+
+            var parser = MockParser.Create(vbe.Object, state);
+            parser.Parse(new CancellationTokenSource());
+            if (parser.State.Status >= ParserState.Error) { Assert.Inconclusive("Parser Error"); }
+
+            project.SetupGet(m => m.VBComponents.Count).Returns(1);
+            vm.SelectedItem = vm.Projects.First();
+
+            Assert.IsTrue(vm.ExportAllCommand.CanExecute(vm.SelectedItem));
+        }
+
+        //[TestCategory("Commands")]
```

Comment:
I need some help with this test.

The idea behind the test is to see if the `CanExecute` works if a empty `CodeExplorerProjectViewModel` is selected.

Due the the parser, etc, the only way to get a `CodeExplorerProjectViewModel` without any child nodes is to create a mock project with a mock component, run a parse, then remove the component and run another parse.  However, I get a `Object reference not set to an instance of an object` exception in `ParseCoordinator.ParseAllInternal`

Or, I can kill it.